GeoLib/Polygon.cpp

```diff
@@ -38,6 +38,25 @@ Polygon::Polygon(Polygon const& other)
 	}
 }
 
+Polygon& Polygon::operator=(Polygon const& other)
+{
+	if (other._ply_pnts != _ply_pnts) {
+		ERR(
+		    "Polygon::operator=: polygon other and this polygon are not based "
+		    "on the same point vector. Aborting");
+		std::abort();
+	}
+	if (other == *this)
+		return *this;
+
+	Polygon tmp(other);
+	std::swap(tmp._aabb, _aabb);
+	std::swap(tmp._simple_polygon_list, _simple_polygon_list);
+	std::swap(tmp._length, _length);
+	std::swap(tmp._ply_pnt_ids, _ply_pnt_ids);
```

Comment:
You only have two data members inside `Polygon`: `_simple_polygon_list` and `_aabb`. Are the other two defined in the base class?
Then it would be better to use `operator=` of `Polyline` for them, e.g., `Polyline::operator=(*this, other);` (I'm not sure if it works like this.) If you think, the code becomes too complicated by that, just leave it as it is.